internal/infrastructure/google/fetch_test.go

```diff
@@ -0,0 +1,90 @@
+package google
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/reearth/reearth-backend/pkg/file"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/h2non/gock.v1"
+)
+
+func Test_fetchCSV(t *testing.T) {
+	t.Cleanup(func() {
+		gock.EnableNetworking()
+		gock.OffAll()
+	})
+
+	gock.DisableNetworking()
+
+	type args struct {
+		token     string
+		fileId    string
+		sheetName string
+	}
+	tests := []struct {
+		name    string
+		setup   func()
+		args    args
+		want    *file.File
+		wantErr bool
+	}{
+		{
+			name: "Invalid Token",
+			setup: func() {
+				gock.New("https://docs.google.com").
+					Get("/spreadsheets/d/(.*)/gviz/tq").
+					PathParam("d", "1bXBDUrOgYWdHzScMiLNHRUsmNC9SUV4VFOvpqrx0Yok").
+					MatchParams(map[string]string{
+						"tqx":   "out:csv",
+						"sheet": "Dataset1",
+					}).
+					Reply(http.StatusUnauthorized)
+			},
+			args: args{
+				token:     "ya29.a0AfH6SMA-glivrhwHjCCMLCTb8ROzUnvNUV73O7P1DLPFmOVRGAKxaozbQamKNAV8Ajd3uEnWdVbIrKUVHzotnlOInreqXgm6gWvG4NWDbyRMR58QHMq49YWs64gzw8V-KxfvUlnECgqN6JIW1uSMUqYBSDzk9g",
+				fileId:    "1bXBDUrOgYWdHzScMiLNHRUsmNC9SUV4VFOvpqrx0Yok",
+				sheetName: "Dataset1",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Working scenario",
+			setup: func() {
+				gock.New("https://docs.google.com").
+					Get("/spreadsheets/d/(.*)/gviz/tq").
+					PathParam("d", "1bXBDUrOgYWdHzScMiLNHRUsmNC9SUV4VFOvpqrx0Yok").
+					MatchParams(map[string]string{
+						"tqx":   "out:csv",
+						"sheet": "Dataset1",
+					}).
+					Reply(http.StatusOK).
+					BodyString("lat,lng,hieght\n30,35,300\n30.1,35,400")
+			},
+			args: args{
+				token:     "ya29.a0AfH6SMA-glivrhwHjCCMLCTb8ROzUnvNUV73O7P1DLPFmOVRGAKxaozbQamKNAV8Ajd3uEnWdVbIrKUVHzotnlOInreqXgm6gWvG4NWDbyRMR58QHMq49YWs64gzw8V-KxfvUlnECgqN6JIW1uSMUqYBSDzk9g",
```

Comment:
Tokens are public. is it safe? 

I understand that HTTP communication is mocked, but it is a good idea to use a string of characters that can be easily identified as a dummy token. Otherwise, someone may try to abuse the token (even if it is a token that cannot be used at the time of the result).

Reply:
Resolved